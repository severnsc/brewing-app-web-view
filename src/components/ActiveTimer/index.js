import React from "react"
import styles from "./styles"
import PropTypes from "prop-types"

const ActiveTimer = ({ name, time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{name}</h2>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	name: PropTypes.string.isRequired,
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer