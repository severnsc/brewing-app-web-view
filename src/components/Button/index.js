import React from "react"
import PropTypes from "prop-types"
import globalStyles from "../styles"

const Button = ({ onClick, value, style }) => (
	<button style={{...globalStyles.button, ...style}} onClick={onClick}>
		{value}
	</button>
)

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number
	]).isRequired,
	style: PropTypes.object
}

export default Button