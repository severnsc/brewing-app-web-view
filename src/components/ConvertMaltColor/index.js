import React from "react"
import PropTypes from "prop-types"
import { MaltColor } from ".."
import { convertColor } from "../../utils"

const ConvertMaltColor = ({ from, to, value }) => {

	const convertedColor = convertColor(from, to, value)

	let srmValue
	if(from === "SRM"){
		srmValue = value
	}else{
		srmValue = convertColor(from, "SRM", value)
	}

	return <MaltColor srmValue={srmValue} value={convertedColor} />

}

ConvertMaltColor.propTypes = {
	from: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	to: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	value: PropTypes.number.isRequired
}

export default ConvertMaltColor