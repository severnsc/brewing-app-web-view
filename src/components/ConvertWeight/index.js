import React from "react"
import PropTypes from "prop-types"
import { Weight } from ".."
import { convertLbsToKg, convertKgToLbs } from "../../utils"

const ConvertWeight = ({ from, to, amount }) => {

	let convertedAmount
	if(from === "imperial" && to === "metric"){
		convertedAmount = convertLbsToKg(amount)
	}

	if(from === "metric" && to === "imperial"){
		convertedAmount = convertKgToLbs(amount)
	}

	return(
		<Weight unit={to} amount={convertedAmount} />
	)
}

ConvertWeight.propTypes = {
	from: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	to: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	amount: PropTypes.number.isRequired
}

export default ConvertWeight