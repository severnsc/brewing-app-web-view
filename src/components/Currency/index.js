import React from "react"
import PropTypes from "prop-types"

const Currency = ({ amount, unit }) => {

	let currencySymbol
	switch(unit){
		case "USD":
			currencySymbol = "$"
			break

		case "GBP":
			currencySymbol = "£"
			break

		case "EUR":
			currencySymbol = "€"
			break

		default:
			currencySymbol = ""
	}

	const parsedAmount = parseFloat(amount).toFixed(2)

	return(
		<span>
			{currencySymbol}{parsedAmount}
		</span>
	)
}

Currency.propTypes = {
	amount: PropTypes.number.isRequired,
	unit: PropTypes.oneOf(["USD", "GBP", "EUR"]).isRequired
}

export default Currency