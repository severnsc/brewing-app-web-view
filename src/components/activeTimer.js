import React from "react"
import PropTypes from "prop-types"

const ActiveTimer = ({ time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const styles = {
		button: {
			color: "white",
			backgroundColor: "grey"
		}
	}

	styles.start = {
		...styles.button,
		backgroundColor: "green"
	}

	styles.stop = {
		...styles.button,
		backgroundColor: "red"
	}

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer