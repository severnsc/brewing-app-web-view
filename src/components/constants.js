export const errorRed = "hsl(353, 85%, 44%)"
export const green = "hsl(167, 90%, 38%)"