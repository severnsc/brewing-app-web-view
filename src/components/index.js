import Timer from "./Timer"
import ForgotPasswordForm from "./ForgotPasswordForm"
import HoverableTableRow from "./HoverableTableRow"
import InventoryItemForm from "./InventoryItemForm"
import Modal from "./Modal"
import NavBar from "./NavBar"
import ResetPasswordForm from "./ResetPasswordForm"
import SearchBar from "./SearchBar"
import SortableTable from "./SortableTable"
import TimerAlertForm from "./TimerAlertForm"
import TimerForm from "./TimerForm"
import InventoryForm from "./InventoryForm"
import FixedNewItemButton from "./FixedNewItemButton"
import NewTimerForm from "./NewTimerForm"
import NewTimerAlertForm from "./NewTimerAlertForm"
import {
	SignupForm,
	LoginForm
} from "./Forms"

export {
	Timer,
	ForgotPasswordForm,
	HoverableTableRow,
	InventoryItemForm,
	LoginForm,
	Modal,
	NavBar,
	ResetPasswordForm,
	SearchBar,
	SignupForm,
	SortableTable,
	TimerAlertForm,
	TimerForm,
	InventoryForm,
	FixedNewItemButton,
	NewTimerForm,
	NewTimerAlertForm
}