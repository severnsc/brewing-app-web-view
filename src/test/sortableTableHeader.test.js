import React from 'react';
import renderer from "react-test-renderer"
import { SortableTableHeader } from "../components"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("sortable table header component", () => {

	it("renders a sortable table header component", () => {
		const sortableTableHeader = renderer.create(
			<SortableTableHeader
				columns={[]}
				onClick={() => {}}
			/>
		)
		const tree = sortableTableHeader.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders a thead", () => {
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={[]}
				onClick={() => {}}
			/>
		)
		expect(sortableTableHeader.find("thead").length).toBe(1)
	})

	it("renders a tr as child of thead", () => {
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={[]}
				onClick={() => {}}
			/>
		)
		expect(sortableTableHeader.find("thead tr").length).toBe(1)
	})

	it("renders a th for each item in the columns array prop", () => {
		const columns = ["column1", "column2"]
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				onClick={() => {}}
			/>
		)
		expect(sortableTableHeader.find("th").length).toBe(2)
	})

	it("sets each th text equal to text of item in columns array prop", () => {
		const columns = ["column1", "column2"]
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				onClick={() => {}}
			/>
		)
		sortableTableHeader.find("th").forEach((n, i) => 
			expect(n.text().includes(columns[i])).toBe(true)
		)
	})

	it("renders a sort indicator in the th whose text equals the sortBy prop", () => {
		const columns = ["column1", "column2"]
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				sortBy="column1"
				onClick={() => {}}
			/>
		)
		expect(sortableTableHeader.findWhere(n => 
			n.text() === "column1 ∧" && n.type() === "th").length
		).toBe(1)
	})

	it("renders a ∧ in the column that equals sortBy if sortOrder is asc", () => {
		const columns = ["column1", "column2"]
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				sortBy="column2"
				sortOrder="asc"
				onClick={() => {}}
			/>
		)
		expect(sortableTableHeader.findWhere(n => 
			n.text() === "column2 ∧" && n.type() === "th").length
		).toBe(1)
	})

	it("renders a ∨ in the column that equals sortBy if sortOrder is desc", () => {
		const columns = ["column1", "column2"]
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				sortBy="column2"
				sortOrder="desc"
				onClick={() => {}}
			/>
		)
		expect(sortableTableHeader.findWhere(n => 
			n.text() === "column2 ∨" && n.type() === "th").length
		).toBe(1)
	})

	it("renders the first item in columns as asc if sortBy and sortOrder are undefined", () => {
		const columns = ["column1", "column2"]
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				onClick={() => {}}
			/>
		)
		const node = sortableTableHeader.findWhere(n => 
			n.text().includes("∧") && n.type() === "th")
		expect(node.text().includes("column1")).toBe(true)
	})

	it("calls the onClick prop on th click", () => {
		const columns = ["column1", "column2"]
		let onClickCalled = false
		const onClick = () => {
			onClickCalled = true
		}
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				onClick={onClick}
			/>
		)
		sortableTableHeader.findWhere(n => 
			n.text().includes("column1") && n.type() === "th").simulate("click")
		expect(onClickCalled).toBe(true)	
	})

	it("passes the clicked column name to the onClick prop on th click", () => {
		const columns = ["column1", "column2"]
		let onClickArg = null
		const onClick = name => {
			onClickArg = name
		}
		const sortableTableHeader = shallow(
			<SortableTableHeader
				columns={columns}
				onClick={onClick}
			/>
		)
		sortableTableHeader.findWhere(n => 
			n.text().includes("column1") && n.type() === "th").simulate("click")
		expect(onClickArg).toBe("column1")	
	})

	describe("styles", () => {

		it("has backgroundColor hsl(0, 0%, 95%) on the thead", () => {
			const sortableTableHeader = shallow(
				<SortableTableHeader
					columns={[]}
					onClick={() => {}}
				/>
			)
			expect(
				sortableTableHeader.find("thead").prop("style").backgroundColor
			).toBe("hsl(0, 0%, 95%)")
		})

		it("has color hsl(0, 0%, 50%) on the thead", () => {
			const sortableTableHeader = shallow(
				<SortableTableHeader
					columns={[]}
					onClick={() => {}}
				/>
			)
			expect(
				sortableTableHeader.find("thead").prop("style").color
			).toBe("hsl(0, 0%, 50%)")
		})

		it("has cursor pointer on the thead", () => {
			const sortableTableHeader = shallow(
				<SortableTableHeader
					columns={[]}
					onClick={() => {}}
				/>
			)
			expect(
				sortableTableHeader.find("thead").prop("style").cursor
			).toBe("pointer")
		})

		it("has padding 20px 10px on each th", () => {
			const columns = ["column1", "column2"]
			const sortableTableHeader = shallow(
				<SortableTableHeader
					columns={columns}
					onClick={() => {}}
				/>
			)
			sortableTableHeader.find("th").forEach(n => 
				expect(n.prop("style").padding).toBe("20px 10px"))
		})

		it("has width equal to 100/columns.length % on each th", () => {
			const columns = ["column1", "column2"]
			const sortableTableHeader = shallow(
				<SortableTableHeader
					columns={columns}
					onClick={() => {}}
				/>
			)
			sortableTableHeader.find("th").forEach(n => 
				expect(n.prop("style").width).toBe(100/columns.length + "%"))
		})

	})

})