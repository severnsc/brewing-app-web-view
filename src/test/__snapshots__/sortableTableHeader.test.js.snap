// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortable table header component renders a sortable table header component 1`] = `
<thead
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 95%)",
      "color": "hsl(0, 0%, 50%)",
      "cursor": "pointer",
    }
  }
>
  <tr />
</thead>
`;
